s2a_options.go

```diff
@@ -121,6 +125,34 @@ type ClientOptions struct {
 	// VerificationMode specifies the mode that S2A must use to verify the
 	// peer certificate chain.
 	VerificationMode VerificationModeType
+
+	// S2Av2 only.
+	// Optional fallback after dialing with S2Av2 fails.
+	FallbackOpts *FallbackOptions
+}
+
+// FallbackOptions contains the fallback options for S2Av2.
+type FallbackOptions struct {
+	// FallbackClientHandshakeFunc is used to specify fallback behavior when calling s2a.NewClientCreds().
+	// It will be called by s2av2TransportCreds's ClientHandshake function, after handshake with S2Av2 fails.
+	//     originServer: the original server attempted with S2Av2.
```

Comment:
I'm afraid this terminology is confusing: could we instead say "serverAddress" or "targetAddress"? Since this is the address of the server we're trying to connect to?

Similarly, could "originConn" be renamed to "conn" or "serverConn".

Using the term "origin" or "original" implies that there is necessarily a difference, when that's not the case (it just happens to be the case for the default implementation).